tools/wake/describe.cpp

```diff
@@ -169,16 +205,53 @@ static void describe_shell(const std::vector<JobReflection> &jobs, bool debug, b
   }
 }
 
-void describe(const std::vector<JobReflection> &jobs, bool script, bool debug, bool verbose,
-              const char *taguri) {
-  if (taguri) {
-    for (auto &job : jobs)
-      for (auto &tag : job.tags)
-        if (tag.uri == taguri) std::cout << tag.content << std::endl;
-  } else if (script) {
-    describe_shell(jobs, debug, verbose);
-  } else {
-    describe_human(jobs, debug, verbose);
+void describe_human(const std::vector<JobReflection> &jobs) {
+  for (auto &job : jobs) {
+    std::cout << "\n# " << job.label << "(" << job.job << ")\n$ ";
+    for (auto &cmd_part : job.commandline) {
+      std::cout << cmd_part << " ";  // # TODO: don't output trailing space
+    }
+    std::cout << "\n\n";
+
+    std::cout << "================================================\n" << std::endl;
+
+    // We have to use our speical stream for the output of the program
+    for (auto &log_line : job.std_writes) {
+      std::cout << log_line.first;
+    }
+
+    std::cout << "------------------------------------------------\n" << std::endl;
+  }
+}
+
+void describe(const std::vector<JobReflection> &jobs, DescribePolicy policy) {
+  switch (policy.type) {
+    case policy.SCRIPT: {
+      describe_shell(jobs, true, true);
+      break;
+    }
+    case policy.HUMAN: {
+      describe_human(jobs);
+      break;
+    }
+    case policy.TAG_URI: {
```

Comment:
I'm 75% sure that you did this right but let me see if I can make a tagged job real quick